### ISSUE TITLE
50/60 fps stuttering on vo gpu, xv works well though (intel).

### ISSUE THREAD

**Issue by eriksl** [mpv-player/mpv#10658]
### Important Information

Provide following Information:
- mpv version: 0.34
- Linux Distribution and Version: Fedora 36
- Source of the mpv binary:
- If known which version of mpv introduced the problem
- Window Manager and version: marco (mate)
- GPU driver and version: unknown, Intel GPU
- Possible screenshot or video of visual glitches

If you're not using git master or the latest release, update.
Releases are listed here: https://github.com/mpv-player/mpv/releases

### Reproduction steps

Try to reproduce your issue with --no-config first. If it isn't reproducible
with --no-config try to first find out which option or script causes your issue.
- does not matter

Describe the reproduction steps as precise as possible. It's very likely that
the bug you experience wasn't reproduced by the developer because the workflow
differs from your own.
- generate video in 1080p50 or 1080p60 and play it. codecs do not matter

### Expected behavior
- smooth playback

### Actual behavior
- playback is stuttering, but there is no framedrop mentioned

### Log file

Make a log file made with -v -v or --log-file=output.txt, paste it to
https://0x0.st/ or attach it to the github issue, and replace this text with a
link to it.

- [log.txt](https://github.com/mpv-player/mpv/files/9599741/log.txt)

The issue will be closed for ignoring the issue template.

### Sample files

Sample files needed to reproduce this issue can be uploaded to https://0x0.st/
or similar sites. (Only needed if the issue cannot be reproduced without it.)
Do not use garbage like "cloud storage", especially not Google Drive.

- Not relevant.

Additional information. I am using a HD Graphics 630 GPU. Two monitors are connected using DP, they have 4K resolution. I was told before that the Intel GPU is too weak to do the scaling from 1080p to 4K in realtime (which I highly doubt, but OK), so I bought me an AMD GPU card "AMD/ATI] Lexa XT [Radeon PRO WX 2100]". This card has several issues (including driver bugs) so I'd rather not use it, but it appears it doesn't even solve the problem. Playback is still stuttering and I can only have smooth playback using the xv vo. Besides that, none of them is able to prevent tearing, in gpu mode, in any way. Only solution, again, is to use the xv vo.

Surely I am not the only one having this setup? Using 4k monitors and trying to play 1080p50/1080p60 clips?

I repeat: there is NO mentioning of framedrops. It's no CPU limitation. The timing is simply wrong (varies, apparently) so the motion is not smooth.

**Comment by eriksl** [comment]
Additional info: when running in "--untimed --no-audio" mode I can see the OSD play time starting to lag behind, so it really can't cope (in "gpu" mode). In xv mode it can almost keep up. I think it's strange it's not reported as frame drops, play time just starts to skew compared to wall clock time.

**Comment by eriksl** [comment]
--video-unscaled=no does not improve things a lot, so I really do not suspect the scaling from 1080p to 4k.

**Comment by philipl** [comment]
Your GPU is old enough that this might well be a pure memory bandwidth problem doing the scaling. You are not doing hardware decoding (can Kaby Lake do 8bit HEVC? maybe?). If it can decode the video in hardware, you should. It will save some memory and CPU bandwidth, and that might be enough to make it usable.

**Comment by eriksl** [comment]
I tried with both hardware decoding on and off.

What puzzles me is that the AMD external GPU has exactly the same issues. For the moment I have it render in the memory of the Intel GPU because it's DP outputs are better, but I will try using connected them natively soon. I seem to remember that didn't really fix the issue either.

Thanks for your interest in the issue.

**Comment by eriksl** [comment]
I just tried again with hardware decoding on both gpu's. No change really.

**Comment by sfan5** [comment]
Tried `--vd-lavc-dr=no`?

**Comment by eriksl** [comment]
I think this is solved now :-)

For future reference and documentation, this is what I did and the results.

1) no external GPU, just Intel. Result: choppy playback of 50/60 fps clips, both windowed and full screen. Compositing on or off no difference.

2) added an external GPU (AMD WX2100). Both cards initialised by the system and both used by X11. Monitors plugged into the AMD GPU. Result: still choppy playback.

3) plugged the monitors into the connectors of the Intel GPU. Still choppy (which was to be expected, but I had to try, because I had issues with the ADM's connectors).

4) added DRI_PRIME=1 to the environment of mpv. Mpv correctly shows it's using the AMD GPU now (-v). Still choppy playback. WTF?

5) went all the way and disabled the Intel GPU in the BIOS settings. The name of the option doesn't even come close to what it does, so that wasn't trivial to find out ;-). Now I only have on GPU, both in hardware, X11, DRI acceleration and mpv. Intel GPU completely gone. **Now the playback is smooth and without any drops**. Even with the compositor enabled.

So this is what I think was happening: as long as the Intel GPU is enabled (and not even used), it's memory is used as frame buffer. This memory is "stolen" main memory which is not as fast as GPU memory by far. Now with the Intel GPU disabled, the AMD GPU is finally using it's own, internal memory. Good to know...

Thanks everybody for joining in.
